src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Data/UserAccessor.cs

```diff
@@ -0,0 +1,28 @@
+using BlazorWeb_CSharp.Data;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Identity;
+
+namespace BlazorWeb_CSharp.Data;
+
+internal sealed class UserAccessor(
+    IHttpContextAccessor httpContextAccessor,
+    UserManager<ApplicationUser> userManager,
+    NavigationManager navigationManager)
+{
+    public async Task<ApplicationUser> GetRequiredUserAsync()
+    {
+        var principal = httpContextAccessor.HttpContext?.User ??
+            throw new InvalidOperationException($"{nameof(GetRequiredUserAsync)} requires access to an {nameof(HttpContext)}.");
```

Comment:
Can we avoid the httpContextAccessor pattern and pass the HttpContext as a parameter instead?

Templates represent "good" code, and we don't want to encourage the usage of HttpContextAccessor if we can avoid it. In this case, its an extra line I think (in most cases)

Reply:
This is why I mentioned it might be nice to have access to access CascadingParameters from normal scoped services that we expect will be used from components rather than only via explicit parameters in the components themselves.

Having to manually `[CascadingParameter] private HttpContext HttpContext { get; set; } = default!;` in every component that needs to get the current user from the UserManager or that may need to redirect is far worse than using `IHttpContextAccessor` for now.

Reply:
@halter73 You can create an `IdentityPageBase` base component, put it there, and make your components inherit from it. You can do that in a `_viewImports.razor` and it it will be available directly on all `.razor` files.

Reply:
While having a base class would work, it is a bit of a constraining solution (people may have their own other base classes they want to use). I'd be in favour of just adding `[CascadingParameter] private required HttpContext HttpContext { get; set; }` wherever it's needed (and I know it's in about 15 places).